src/code/z_bgcheck.c

```diff
@@ -7,44 +7,51 @@
 #define DYNA_RAYCAST_WALLS 2
 #define DYNA_RAYCAST_CEILINGS 4
 
-u32 sWallFlags[32] = {
-    0, 1, 3, 5, 8, 16, 32, 64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+u32 sWallFlags[BG_WALL_TYPE_MAX] = {
+    0,                         // BG_WALL_TYPE_0
+    WALL_FLAG_0,               // BG_WALL_TYPE_1
+    WALL_FLAG_0 | WALL_FLAG_1, // BG_WALL_TYPE_2
+    WALL_FLAG_0 | WALL_FLAG_2, // BG_WALL_TYPE_3
+    WALL_FLAG_3,               // BG_WALL_TYPE_4
+    WALL_FLAG_4,               // BG_WALL_TYPE_5
+    WALL_FLAG_5,               // BG_WALL_TYPE_6
+    WALL_FLAG_6,               // BG_WALL_TYPE_7
 };
 
-u16 sSurfaceTypeSfx[] = {
-    /* 0x00 */ NA_SE_PL_WALK_GROUND - SFX_FLAG,
-    /* 0x01 */ NA_SE_PL_WALK_SAND - SFX_FLAG,
-    /* 0x02 */ NA_SE_PL_WALK_CONCRETE - SFX_FLAG,
-    /* 0x03 */ NA_SE_PL_WALK_DIRT - SFX_FLAG,
-    /* 0x04 */ NA_SE_PL_WALK_WATER0 - SFX_FLAG,
-    /* 0x05 */ NA_SE_PL_WALK_WATER1 - SFX_FLAG,
-    /* 0x06 */ NA_SE_PL_WALK_WATER2 - SFX_FLAG,
-    /* 0x07 */ NA_SE_PL_WALK_MAGMA - SFX_FLAG,
-    /* 0x08 */ NA_SE_PL_WALK_GRASS - SFX_FLAG,
-    /* 0x09 */ NA_SE_PL_WALK_GLASS - SFX_FLAG,
-    /* 0x0A */ NA_SE_PL_WALK_LADDER - SFX_FLAG,
-    /* 0x0B */ NA_SE_PL_WALK_GROUND - SFX_FLAG,
-    /* 0x0C */ NA_SE_PL_WALK_ICE - SFX_FLAG,
-    /* 0x0D */ NA_SE_PL_WALK_IRON - SFX_FLAG,
-    /* 0x0E */ NA_SE_PL_WALK_SNOW - SFX_FLAG
+u16 sSurfaceTypeSfx[BG_SURFACE_SFX_TYPE_MAX] = {
+    NA_SE_PL_WALK_GROUND - SFX_FLAG,   // BG_SURFACE_SFX_TYPE_0
+    NA_SE_PL_WALK_SAND - SFX_FLAG,     // BG_SURFACE_SFX_TYPE_1
+    NA_SE_PL_WALK_CONCRETE - SFX_FLAG, // BG_SURFACE_SFX_TYPE_2
+    NA_SE_PL_WALK_DIRT - SFX_FLAG,     // BG_SURFACE_SFX_TYPE_3
+    NA_SE_PL_WALK_WATER0 - SFX_FLAG,   // BG_SURFACE_SFX_TYPE_4
+    NA_SE_PL_WALK_WATER1 - SFX_FLAG,   // BG_SURFACE_SFX_TYPE_5
+    NA_SE_PL_WALK_WATER2 - SFX_FLAG,   // BG_SURFACE_SFX_TYPE_6
+    NA_SE_PL_WALK_MAGMA - SFX_FLAG,    // BG_SURFACE_SFX_TYPE_7
+    NA_SE_PL_WALK_GRASS - SFX_FLAG,    // BG_SURFACE_SFX_TYPE_8
+    NA_SE_PL_WALK_GLASS - SFX_FLAG,    // BG_SURFACE_SFX_TYPE_9
+    NA_SE_PL_WALK_LADDER - SFX_FLAG,   // BG_SURFACE_SFX_TYPE_10
+    NA_SE_PL_WALK_GROUND - SFX_FLAG,   // BG_SURFACE_SFX_TYPE_11
+    NA_SE_PL_WALK_ICE - SFX_FLAG,      // BG_SURFACE_SFX_TYPE_12
+    NA_SE_PL_WALK_IRON - SFX_FLAG,     // BG_SURFACE_SFX_TYPE_13
+    NA_SE_PL_WALK_SNOW - SFX_FLAG,     // BG_SURFACE_SFX_TYPE_14
 };
 
-u8 D_801B46C0[] = {
-    /* 0x00 */ 1,
-    /* 0x01 */ 1,
-    /* 0x02 */ 0,
-    /* 0x03 */ 1,
-    /* 0x04 */ 0,
-    /* 0x05 */ 0,
-    /* 0x06 */ 0,
-    /* 0x07 */ 0,
-    /* 0x08 */ 0,
-    /* 0x09 */ 0,
-    /* 0x0A */ 0,
-    /* 0x0B */ 0,
-    /* 0x0C */ 0,
-    /* 0x0D */ 0,
-    /* 0x0E */ 1
+u8 D_801B46C0[BG_SURFACE_SFX_TYPE_MAX] = {
+    true,  // BG_SURFACE_SFX_TYPE_0
+    true,  // BG_SURFACE_SFX_TYPE_1
+    false, // BG_SURFACE_SFX_TYPE_2
```

Comment:
I'm not sure that it's correct to express these in terms of `true`/`false`, given that the data is read as being a set of flags